module/zfs/zpl_file.c

```diff
@@ -888,6 +888,38 @@ zpl_ioctl_setxattr(struct file *filp, void __user *arg)
 	return (err);
 }
 
+static long
+zpl_ioctl_count_filled(struct file *filep, void __user *arg)
```

Comment:
For the most part we've tried to keep the Linux specific changes confined to the `zpl_*` functions, and left the core ZFS functionality in `zfs_vnops.c`, `zfs_vfsops.c`, etc.  So let's go ahead and split this function up the same way.  Lines 901-914 can be moved in to a `zfs_count_filled()` function in `zfs_vnops.c` and then `zpl_ioctl_count_filled` is what's left.  This way _most_ of the linux data structures, functions, negative return codes don't seep down very far in to ZFS proper.

```c
static long
zpl_ioctl_count_filled(struct file *filep, void __user *arg)
{
	struct inode *ip = file_inode(filep);
	zfsvfs_t *zfsvfs = ITOZSB(ip);
	dmu_object_info_t doi;
	int error;

        error = -zfs_count_filled(zfsvfs, ITOZ(ip), &doi);
        if (error == 0) {
        	uint64_t ndata = doi.doi_fill_count;
        	error = copy_to_user(arg, &ndata, sizeof (ndata));
        }

	return (error);
}
```

Reply:
Makes sense.  I adjusted the interface you suggested slightly:
```c
int zfs_count_filled(struct inode *ip, uint64_t *fill_count)
```